src/test/java/com/teragrep/cfe_35/router/NoSuchTargetTest.java

```diff
@@ -90,22 +94,43 @@ private void setup(int port, List<byte[]> recordList) throws IOException {
         Consumer<FrameContext> cbFunction = relpFrameServerRX -> recordList
                 .add(relpFrameServerRX.relpFrame().payload().toBytes());
 
+        EventLoopFactory eventLoopFactory = new EventLoopFactory();
+        EventLoop eventLoop = eventLoopFactory.create(); // FIXME this is not cleaned up
+        Thread eventLoopThread = new Thread(eventLoop);
+        eventLoopThread.start(); // FIXME this is not cleaned up
+
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         ServerFactory serverFactory = new ServerFactory(
+                eventLoop,
                 executorService,
                 new PlainFactory(),
-                () -> new DefaultFrameDelegate(cbFunction)
+                new FrameDelegationClockFactory(() -> new DefaultFrameDelegate(cbFunction))
         );
-        Server server = serverFactory.create(port);
-        Thread serverThread = new Thread(server);
-        serverThread.start();
+        serverFactory.create(port);
     }
 
     @Test
     public void noSuchTargetTest() throws IOException {
         System.setProperty("routingTargetsConfig", "src/test/resources/targetsNoSuchTarget.json");
         RoutingConfig routingConfig = new RoutingConfig();
-        try (TargetRouting targetRouting = new ParallelTargetRouting(routingConfig, this.metricRegistry)) {
+
+        EventLoopFactory eventLoopFactory = new EventLoopFactory();
+        EventLoop eventLoop = eventLoopFactory.create(); // FIXME this is not cleaned up
+        Thread eventLoopThread = new Thread(eventLoop);
+        eventLoopThread.start(); // FIXME this is not cleaned up
+
+        ExecutorService executorService = Executors.newFixedThreadPool(4); // FIXME this is not cleaned up
+
+        ConnectContextFactory connectContextFactory = new ConnectContextFactory(executorService, new PlainFactory());
+        RelpClientFactory clientFactory = new RelpClientFactory(connectContextFactory, eventLoop);
+
+        try (
```

Comment:
I'd prefer try-with-resources to not be used in tests and just manually close it if necessary